internal/driver/glfw/loop.go

```diff
@@ -55,16 +48,8 @@ func runOnMain(f func()) {
 }
 
 // force a function f to run on the draw thread
-func runOnDraw(w *window, f func()) {
-	if drawOnMainThread {
-		runOnMain(func() { w.RunWithContext(f) })
-		return
-	}
-	done := common.DonePool.Get()
-	defer common.DonePool.Put(done)
-
-	drawFuncQueue <- drawData{f: f, win: w, done: done}
-	<-done
+func runOnMainWithContext(w *window, f func()) {
+	runOnMain(func() { w.RunWithContext(f) }) // TODO remove this completely
```

Comment:
Is this a TODO for an upcoming PR or should it be removed?

Reply:
Yes it is part of the future refactoring. Lots more to be done as mentioned above and in the other tickets.

There are 3 different ways to queue onto what is essentially the same thread but I didn't want to rip out too much code until the direction is established :)

Reply:
Cool. Just wanted to check. Still approved from my end :)